### ISSUE TITLE
nvidia-container-cli attempts to mount the wrong /proc/driver/nvidia/gpus path

### ISSUE THREAD

**Issue by ernstae** [NVIDIA/nvidia-docker#816]
The Bus Location of my K80 GPU is 3e3f8e05:00:00.0 and the appropriate tree exists at that location

```
# tree /proc/driver/nvidia/gpus
/proc/driver/nvidia/gpus
└── 3e3f8e05:00:00.
    ├── information
    └── registry

1 directory, 2 files
```

When I attempt to start a container, with the `nvidia-docker2` runtime, the runtime is unable to mount `/proc/driver/nvidia/gpus/8e05:00:00.0:` *take note, that this is a truncated path that drops the first four characters `3e3f` from the device id.

*This behavior has been observed in multiple versions of the nvidia-kmod driver*

The error received is:
```
# docker run --runtime=nvidia nvidia/cuda:9.2-base-centos7 nvidia-smi
Unable to find image 'nvidia/cuda:9.2-base-centos7' locally
9.2-base-centos7: Pulling from nvidia/cuda
256b176beaff: Already exists
c06029c439c8: Pull complete
c001ebb61ce4: Pull complete
6d43ce36a01d: Pull complete
24e518b6fa38: Pull complete
Digest: sha256:53c9614b0a8549af55114386db7cdc10506787f7c05be4a75b553381f0c6529f
Status: Downloaded newer image for nvidia/cuda:9.2-base-centos7
docker: Error response from daemon: OCI runtime create failed: container_linux.go:348: starting container process caused "process_linux.go:402: container init caused \"process_linux.go:385: running prestart hook 1 caused \\\"error running hook: exit status 1, stdout: , stderr: exec command: [/usr/bin/nvidia-container-cli --load-kmods configure --ldconfig=@/sbin/ldconfig --device=all --compute --utility --require=cuda>=9.2 --pid=34877 /var/lib/docker/overlay2/c23e6783532275110450baef2d04b45ec5f911a52b47dbaf30842c4d2692d314/merged]\\\\nnvidia-container-cli: mount error: stat failed: /proc/driver/nvidia/gpus/8e05:00:00.0: no such file or directory\\\\n\\\"\"": unknown.
ERRO[0010] error waiting for container: context canceled
```

I've tested multiple nvidia-kmod versions (396.44, 396.37, 390.30) on CentOS 7.5
```
# uname -a
Linux **redacted.hostname** 3.10.0-862.11.6.el7.x86_64 #1 SMP Tue Aug 14 21:49:04 UTC 2018 x86_64 x86_64 x86_64 GNU/Linux
```

This is dmesg output while testing:
```
[Sat Sep  1 05:46:00 2018] nvidia 3e3f8e05:00:00.0: can't derive routing for PCI INT A
[Sat Sep  1 05:46:00 2018] nvidia 3e3f8e05:00:00.0: PCI INT A: no GSI
[Sat Sep  1 05:46:00 2018] nvidia-modeset: Loading NVIDIA Kernel Mode Setting Driver for UNIX platforms  396.37  Tue Jun 12 13:35:23 PDT 2018
[Sat Sep  1 05:46:01 2018] [drm] [nvidia-drm] [GPU ID 0x8e050000] Loading driver
[Sat Sep  1 05:46:01 2018] [drm] Initialized nvidia-drm 0.0.0 20160202 for 3e3f8e05:00:00.0 on minor 0
[Sat Sep  1 05:49:12 2018] nvidia-uvm: Loaded the UVM driver in 8 mode, major device number 240
[Sat Sep  1 05:51:43 2018] nvidia 3e3f8e05:00:00.0: irq 24 for MSI/MSI-X
[Sat Sep  1 05:52:01 2018] nvidia 3e3f8e05:00:00.0: irq 24 for MSI/MSI-X
[Sat Sep  1 06:01:14 2018] nvidia 3e3f8e05:00:00.0: irq 24 for MSI/MSI-X
[Sat Sep  1 06:01:41 2018] nvidia 3e3f8e05:00:00.0: irq 24 for MSI/MSI-X
[Sat Sep  1 06:03:57 2018] nvidia 3e3f8e05:00:00.0: irq 24 for MSI/MSI-X
[Sat Sep  1 06:04:20 2018] nvidia 3e3f8e05:00:00.0: irq 24 for MSI/MSI-X
[Sat Sep  1 06:04:38 2018] nvidia 3e3f8e05:00:00.0: irq 24 for MSI/MSI-X
[Sat Sep  1 06:11:05 2018] nvidia 3e3f8e05:00:00.0: irq 24 for MSI/MSI-X
[Sat Sep  1 06:11:05 2018] docker0: port 1(veth33872d0) entered forwarding state
[Sat Sep  1 06:11:09 2018] docker0: port 1(veth33872d0) entered disabled state
[Sat Sep  1 06:11:10 2018] docker0: port 1(veth33872d0) entered disabled state
[Sat Sep  1 06:11:10 2018] docker0: port 1(veth33872d0) entered disabled state
```

nvidia-smi -a output:
```
root@qus5uw2laif001:~# nvidia-smi -a

==============NVSMI LOG==============

Timestamp                           : Sat Sep  1 06:15:25 2018
Driver Version                      : 396.37

Attached GPUs                       : 1
GPU 3E3F8E05:00:00.0
    Product Name                    : Tesla K80
    Product Brand                   : Tesla
    Display Mode                    : Disabled
    Display Active                  : Disabled
    Persistence Mode                : Disabled
    Accounting Mode                 : Disabled
    Accounting Mode Buffer Size     : 4000
    Driver Model
        Current                     : N/A
        Pending                     : N/A
    Serial Number                   : 0324616218410
    GPU UUID                        : GPU-daade63e-6b82-4c36-32c5-8929377b9e2b
    Minor Number                    : 0
    VBIOS Version                   : 80.21.1F.00.06
    MultiGPU Board                  : Yes
    Board ID                        : 0x8e050000
    GPU Part Number                 : 900-22080-0300-000
    Inforom Version
        Image Version               : 2080.0200.00.04
        OEM Object                  : 1.1
        ECC Object                  : 3.0
        Power Management Object     : N/A
    GPU Operation Mode
        Current                     : N/A
        Pending                     : N/A
    GPU Virtualization Mode
        Virtualization mode         : Pass-Through
    IBMNPU
        Relaxed Ordering Mode       : N/A
    PCI
        Bus                         : 0x00
        Device                      : 0x00
        Domain                      : 0x3E3F8E05
        Device Id                   : 0x102D10DE
        Bus Id                      : 3E3F8E05:00:00.0
        Sub System Id               : 0x106C10DE
        GPU Link Info
            PCIe Generation
                Max                 : 3
                Current             : 3
            Link Width
                Max                 : 16x
                Current             : 16x
        Bridge Chip
            Type                    : N/A
            Firmware                : N/A
        Replays since reset         : 0
        Tx Throughput               : N/A
        Rx Throughput               : N/A
    Fan Speed                       : N/A
    Performance State               : P0
    Clocks Throttle Reasons
        Idle                        : Not Active
        Applications Clocks Setting : Not Active
        SW Power Cap                : Not Active
        HW Slowdown                 : Not Active
            HW Thermal Slowdown     : N/A
            HW Power Brake Slowdown : N/A
        Sync Boost                  : Not Active
        SW Thermal Slowdown         : Not Active
    FB Memory Usage
        Total                       : 11441 MiB
        Used                        : 0 MiB
        Free                        : 11441 MiB
    BAR1 Memory Usage
        Total                       : 16384 MiB
        Used                        : 2 MiB
        Free                        : 16382 MiB
    Compute Mode                    : Default
    Utilization
        Gpu                         : 1 %
        Memory                      : 0 %
        Encoder                     : 0 %
        Decoder                     : 0 %
    Encoder Stats
        Active Sessions             : 0
        Average FPS                 : 0
        Average Latency             : 0
    Ecc Mode
        Current                     : Enabled
        Pending                     : Enabled
    ECC Errors
        Volatile
            Single Bit
                Device Memory       : 0
                Register File       : 0
                L1 Cache            : 0
                L2 Cache            : 0
                Texture Memory      : 0
                Texture Shared      : N/A
                CBU                 : N/A
                Total               : 0
            Double Bit
                Device Memory       : 0
                Register File       : 0
                L1 Cache            : 0
                L2 Cache            : 0
                Texture Memory      : 0
                Texture Shared      : N/A
                CBU                 : N/A
                Total               : 0
        Aggregate
            Single Bit
                Device Memory       : 12
                Register File       : 0
                L1 Cache            : 0
                L2 Cache            : 0
                Texture Memory      : 0
                Texture Shared      : N/A
                CBU                 : N/A
                Total               : 12
            Double Bit
                Device Memory       : 17
                Register File       : 0
                L1 Cache            : 0
                L2 Cache            : 0
                Texture Memory      : 0
                Texture Shared      : N/A
                CBU                 : N/A
                Total               : 17
    Retired Pages
        Single Bit ECC              : 0
        Double Bit ECC              : 4
        Pending                     : No
    Temperature
        GPU Current Temp            : 40 C
        GPU Shutdown Temp           : 110 C
        GPU Slowdown Temp           : 88 C
        GPU Max Operating Temp      : N/A
        Memory Current Temp         : N/A
        Memory Max Operating Temp   : N/A
    Power Readings
        Power Management            : Supported
        Power Draw                  : 79.76 W
        Power Limit                 : 149.00 W
        Default Power Limit         : 149.00 W
        Enforced Power Limit        : 149.00 W
        Min Power Limit             : 100.00 W
        Max Power Limit             : 175.00 W
    Clocks
        Graphics                    : 875 MHz
        SM                          : 875 MHz
        Memory                      : 2505 MHz
        Video                       : 540 MHz
    Applications Clocks
        Graphics                    : 562 MHz
        Memory                      : 2505 MHz
    Default Applications Clocks
        Graphics                    : 562 MHz
        Memory                      : 2505 MHz
    Max Clocks
        Graphics                    : 875 MHz
        SM                          : 875 MHz
        Memory                      : 2505 MHz
        Video                       : 540 MHz
    Max Customer Boost Clocks
        Graphics                    : N/A
    Clock Policy
        Auto Boost                  : On
        Auto Boost Default          : On
    Processes                       : None
```

Packages installed:
```
# rpm -qa '*nvidia*'
xorg-x11-drv-nvidia-396.37-1.el7.x86_64
libnvidia-container1-1.0.0-0.1.rc.2.x86_64
nvidia-container-runtime-2.0.0-1.docker18.06.1.x86_64
libnvidia-container-tools-1.0.0-0.1.rc.2.x86_64
nvidia-kmod-396.37-2.el7.x86_64
nvidia-docker2-2.0.3-1.docker18.06.1.ce.noarch
xorg-x11-drv-nvidia-diagnostic-396.37-1.el7.x86_64
nvidia-container-runtime-hook-1.4.0-1.x86_64
xorg-x11-drv-nvidia-devel-396.37-1.el7.x86_64
xorg-x11-drv-nvidia-libs-396.37-1.el7.x86_64
xorg-x11-drv-nvidia-gl-396.37-1.el7.x86_64
```

 - [ ] NVIDIA container library version from `nvidia-container-cli -V`
```
# nvidia-container-cli -V
version: 1.0.0
build date: 2018-06-12T00:20+0000
build revision: e3a2035da5a44b8a83d9568b91a8a0b542ee15d5
build compiler: gcc 4.8.5 20150623 (Red Hat 4.8.5-28)
build platform: x86_64
build flags: -D_GNU_SOURCE -D_FORTIFY_SOURCE=2 -DNDEBUG -std=gnu11 -O2 -g -fdata-sections -ffunction-sections -fstack-protector -fno-strict-aliasing -fvisibility=hidden -Wall -Wextra -Wcast-align -Wpointer-arith -Wmissing-prototypes -Wnonnull -Wwrite-strings -Wlogical-op -Wformat=2 -Wmissing-format-attribute -Winit-self -Wshadow -Wstrict-prototypes -Wunreachable-code -Wconversion -Wsign-conversion -Wno-unknown-warning-option -Wno-format-extra-args -Wno-gnu-alignof-expression -Wl,-zrelro -Wl,-znow -Wl,-zdefs -Wl,--gc-sections
```

 - [ ] NVIDIA container library logs (see [troubleshooting](https://github.com/NVIDIA/nvidia-docker/wiki/Troubleshooting))

 - [ ] Docker command, image and tag used
```
docker run --runtime=nvidia nvidia/cuda:9.2-base-centos7 nvidia-smi
docker run --runtime=nvidia nvidia/cuda:9.1-base-centos7 nvidia-smi
```

**Comment by ernstae** [comment]
This *may* be related to Microsoft Azure NC6 instance types and their implementation.
`lspci -D` reveals the full 8-character domain id, which is potentially the root of this issue.

```
$ lspci -D | grep NVIDIA
3e3f8e05:00:00.0 3D controller: NVIDIA Corporation GK210GL [Tesla K80] (rev a1)
```

From the `lspci` man page, *domains  are numbered  from  0 to ffff* which suggests that this is an issue further upstream.

**Comment by 3XX0** [comment]
Where did you get the drivers from?

**Comment by ernstae** [comment]
Hi there @3XX0  - the drivers came from the NVIDIA RPM yum repo...

```
Name        : nvidia-kmod
Epoch       : 1
Version     : 396.44
Release     : 2.el7
Architecture: x86_64
Install Date: (not installed)
Group       : System Environment/Kernel
Size        : 25041657
License     : Redistributable, no modification permitted
Signature   : RSA/SHA512, Thu 12 Jul 2018 02:49:56 AM UTC, Key ID f60f4b3d7fa2af80
Source RPM  : nvidia-kmod-396.44-2.el7.src.rpm
Build Date  : Thu 12 Jul 2018 02:49:48 AM UTC
Build Host  : eris-l64-build049.nvidia.com
Relocations : (not relocatable)
URL         : http://www.nvidia.com/
Summary     : NVIDIA driver kernel modules
Description :
The nvidia 396.44 driver kernel modules.
```

It looks like the domain_id portion of the PCI address is 32-bits instead of 16-bits, and have been told that there may be a fix in the latest/latest Hyper-V drivers for Linux - or in later kernel versions.  It seems that the nvidia drivers can handle the 32-bit domain_id in the address, but the nvidia-docker-plugin doesn't, and is seemingly taking the last 16-bits as the address (while I believe the behavior of other packages, is to take the leading 16-bits).  That statement should be taken with a grain of salt, since I haven't spent enough time digging into the specifics to say that with 100% confidence.

I'm told that there's a newer release of the hyper-v drivers that address the issue for CentOS 7.5, though I'm working on getting my hands on them to verify.

**Comment by 3XX0** [comment]
There has been some changes although last time I checked, it didn't affect the official Tesla drivers.

Did you try with the official [driver runfile](https://www.nvidia.com/drivers/results/136954) or the official [Azure VMI](https://docs.nvidia.com/ngc/ngc-azure-vmi-release-notes/index.html) see if you could reproduce? 

In any case we should have a fix for it soon.

**Comment by ernstae** [comment]
This is my full stack of modules:

```
ernstae@server:~$ lspci
lspci: sysfs_scan: Invalid domain c0d32321

ernstae@server:~$ modinfo hv_vmbus pci-hyperv
filename:       /lib/modules/3.10.0-862.11.6.el7.x86_64/extra/microsoft-hyper-v/hv_vmbus.ko
version:        4.2.6
license:        GPL
retpoline:      Y
rhelversion:    7.5
srcversion:     581BFCB32D09E7A1B019591
alias:          acpi*:VMBus:*
alias:          acpi*:VMBUS:*
depends:
vermagic:       3.10.0-862.11.6.el7.x86_64 SMP mod_unload modversions
parm:           affinity_mode:vmbus channel cpu affinity mode: 0, 1 (int)
filename:       /lib/modules/3.10.0-862.11.6.el7.x86_64/extra/microsoft-hyper-v/pci-hyperv.ko
alias:          vmbus:1df6c444444400449d52802e27ede19f
version:        4.2.6
description:    Microsoft Hyper-V PCI driver
license:        GPL
retpoline:      Y
rhelversion:    7.5
srcversion:     31734EDECAC52733953DA10
alias:          vmbus:1df6c444444400449d52802e27ede19f
depends:        hv_vmbus
vermagic:       3.10.0-862.11.6.el7.x86_64 SMP mod_unload modversions


ernstae@server:~$ nvidia-smi
Wed Sep  5 18:21:02 2018
+-----------------------------------------------------------------------------+
| NVIDIA-SMI 396.44                 Driver Version: 396.44                    |
|-------------------------------+----------------------+----------------------+
| GPU  Name        Persistence-M| Bus-Id        Disp.A | Volatile Uncorr. ECC |
| Fan  Temp  Perf  Pwr:Usage/Cap|         Memory-Usage | GPU-Util  Compute M. |
|===============================+======================+======================|
|   0  Tesla K80           Off  | C0D32321:00:00.0 Off |                  Off |
| N/A   43C    P0    58W / 149W |      0MiB / 12206MiB |      0%      Default |
+-------------------------------+----------------------+----------------------+

+-----------------------------------------------------------------------------+
| Processes:                                                       GPU Memory |
|  GPU       PID   Type   Process name                             Usage      |
|=============================================================================|
|  No running processes found                                                 |
+-----------------------------------------------------------------------------+

ernstae@server:~$ sudo docker run --runtime=nvidia nvidia/cuda nvidia-smi
[sudo] password for ernsta:
docker: Error response from daemon: OCI runtime create failed: container_linux.go:348: starting container process caused "process_linux.go:402: container init caused \"process_linux.go:385: running prestart hook 1 caused \\\"error running hook: exit status 1, stdout: , stderr: exec command: [/usr/bin/nvidia-container-cli --load-kmods configure --ldconfig=@/sbin/ldconfig --device=all --compute --utility --require=cuda>=9.0 --pid=110524 /var/lib/docker/overlay2/d887b6f1676da9422835479a2b61303747be4e8cfcc6f21114196b149bb37f34/merged]\\\\nnvidia-container-cli: mount error: stat failed: /proc/driver/nvidia/gpus/2321:00:00.0: no such file or directory\\\\n\\\"\"": unknown.
ERRO[0005] error waiting for container: context canceled

ernstae@server:~$ uname -a
Linux server 3.10.0-862.11.6.el7.x86_64 #1 SMP Tue Aug 14 21:49:04 UTC 2018 x86_64 x86_64 x86_64 GNU/Linux

ernstae@server:~$ modinfo nvidia
filename:       /lib/modules/3.10.0-862.11.6.el7.x86_64/extra/nvidia.ko.xz
alias:          char-major-195-*
version:        396.37
supported:      external
license:        NVIDIA
retpoline:      Y
rhelversion:    7.5
srcversion:     CDDF5F6229BAD6360A195B1
alias:          pci:v000010DEd00000E00sv*sd*bc04sc80i00*
alias:          pci:v000010DEd*sv*sd*bc03sc02i00*
alias:          pci:v000010DEd*sv*sd*bc03sc00i00*
depends:        ipmi_msghandler,i2c-core
vermagic:       3.10.0-862.11.6.el7.x86_64 SMP mod_unload modversions
parm:           NVreg_Mobile:int
parm:           NVreg_ResmanDebugLevel:int
parm:           NVreg_RmLogonRC:int
parm:           NVreg_ModifyDeviceFiles:int
parm:           NVreg_DeviceFileUID:int
parm:           NVreg_DeviceFileGID:int
parm:           NVreg_DeviceFileMode:int
parm:           NVreg_UpdateMemoryTypes:int
parm:           NVreg_InitializeSystemMemoryAllocations:int
parm:           NVreg_UsePageAttributeTable:int
parm:           NVreg_MapRegistersEarly:int
parm:           NVreg_RegisterForACPIEvents:int
parm:           NVreg_CheckPCIConfigSpace:int
parm:           NVreg_EnablePCIeGen3:int
parm:           NVreg_EnableMSI:int
parm:           NVreg_TCEBypassMode:int
parm:           NVreg_UseThreadedInterrupts:int
parm:           NVreg_EnableStreamMemOPs:int
parm:           NVreg_EnableBacklightHandler:int
parm:           NVreg_EnableUserNUMAManagement:int
parm:           NVreg_MemoryPoolSize:int
parm:           NVreg_IgnoreMMIOCheck:int
parm:           NVreg_RegistryDwords:charp
parm:           NVreg_RegistryDwordsPerDevice:charp
parm:           NVreg_RmMsg:charp
parm:           NVreg_AssignGpus:charp
```

Not sure if that helps, but I'm hoping between Microsoft and NVIDIA having awareness, this can be resolved, with either the NVIDIA Docker plugin can cope with the 32-bit domain id, or for MSFT to represent it as only a 16-bit 😄

**Comment by ernstae** [comment]
I think I have everything sorted with the latest Microsoft hyper-v kernel modules, which present the 16-bit domain_id
Not sure how you want to handle the resolution of this issue - but I believe I've managed to work around the issue for now.

**Comment by ernstae** [comment]
The update to kernel `3.10.0-862.14.4.el7.x86_64` brings this problem back with a vengeance, and using drivers `410.48` and Tesla M60
I'll head back to Microsoft Support but just wanted to let you know that I'm still seeing this as a problem with the new drivers:

```
:~$ sudo nvidia-smi
Tue Oct  2 23:04:52 2018
+-----------------------------------------------------------------------------+
| NVIDIA-SMI 410.48                 Driver Version: 410.48                    |
|-------------------------------+----------------------+----------------------+
| GPU  Name        Persistence-M| Bus-Id        Disp.A | Volatile Uncorr. ECC |
| Fan  Temp  Perf  Pwr:Usage/Cap|         Memory-Usage | GPU-Util  Compute M. |
|===============================+======================+======================|
|   0  Tesla M60           Off  | A45D8951:00:00.0 Off |                  Off |
| N/A   37C    P0    38W / 150W |      0MiB /  8129MiB |      0%      Default |
+-------------------------------+----------------------+----------------------+
|   1  Tesla M60           Off  | A45DAB2B:00:00.0 Off |                  Off |
| N/A   40C    P0    37W / 150W |      0MiB /  8129MiB |      0%      Default |
+-------------------------------+----------------------+----------------------+
|   2  Tesla M60           Off  | A45DBE9E:00:00.0 Off |                  Off |
| N/A   42C    P0    38W / 150W |      0MiB /  8129MiB |      0%      Default |
+-------------------------------+----------------------+----------------------+
|   3  Tesla M60           Off  | A45DCCBD:00:00.0 Off |                  Off |
| N/A   35C    P0    37W / 150W |      0MiB /  8129MiB |      4%      Default |
+-------------------------------+----------------------+----------------------+

+-----------------------------------------------------------------------------+
| Processes:                                                       GPU Memory |
|  GPU       PID   Type   Process name                             Usage      |
|=============================================================================|
|  No running processes found                                                 |
+-----------------------------------------------------------------------------+
user@redactedhost:~$ sudo docker run --runtime=nvidia nvidia/cuda nvidia-smi
docker: Error response from daemon: OCI runtime create failed: container_linux.go:348: starting container process caused "process_linux.go:402: container init caused \"process_linux.go:385: running prestart hook 1 caused \\\"error running hook: exit status 1, stdout: , stderr: exec command: [/usr/bin/nvidia-container-cli --load-kmods configure --ldconfig=@/sbin/ldconfig --device=all --compute --utility --require=cuda>=9.0 --pid=17849 /var/lib/docker/overlay2/ae68ef2ce73ad509cb2e369e5766157abce607bc43c280b0c2d11fc8f776e204/merged]\\\\nnvidia-container-cli: mount error: stat failed: /proc/driver/nvidia/gpus/8951:00:00.0: no such file or directory\\\\n\\\"\"": unknown.
ERRO[0006] error waiting for container: context canceled
```

Obviously, still using the last four digits of the Bus-ID instead of the first four.

**Comment by 3XX0** [comment]
I already filed a bug against our driver internally. Unfortunately I don't have any ETA yet. I will update the issue once it's fixed

**Comment by Bruslan** [comment]
I have the same Bug with my Tesla M60, cent Os 7.5
The nvidia Tesla drivers were installed via the installation file from the nvidia page
https://www.nvidia.com/Download/driverResults.aspx/139178/en-us

Is there any workaround yet?
```
[root@centosgpuNV6 nvidia_driver]# docker run --privileged -it --rm docker.io/mirrorgooglecontainers/cuda-vector-add:v0.1
Unable to find image 'docker.io/mirrorgooglecontainers/cuda-vector-add:v0.1' locally
Trying to pull repository docker.io/mirrorgooglecontainers/cuda-vector-add ...
v0.1: Pulling from docker.io/mirrorgooglecontainers/cuda-vector-add
5d9a20cbabf3: Pull complete
84b2e9f421b6: Pull complete
6f94649104a2: Pull complete
6c16e819a84a: Pull complete
9822cda4c699: Pull complete
1bc138ea32ad: Pull complete
ade909bfe2a5: Pull complete
e70e5ba470d6: Pull complete
ab71e6b7eb90: Pull complete
925740434ebd: Pull complete
2f93605342b5: Pull complete
fe61ad4992f7: Pull complete
Digest: sha256:0705cd690bc0abf54c0f0489d82bb846796586e9d087e9a93b5794576a456aea
Status: Downloaded newer image for docker.io/mirrorgooglecontainers/cuda-vector-add:v0.1
/usr/bin/docker-current: Error response from daemon: oci runtime error: container_linux.go:247: starting container process caused "process_linux.go:339: running prestart hook 1 caused \"error running hook: exit status 1, stdout: , stderr: exec command: [/usr/bin/nvidia-container-cli --load-kmods configure --ldconfig=@/sbin/ldconfig --device=all --compute --compat32 --graphics --utility --video --display --require=cuda>=8.0 --pid=35303 /var/lib/docker/overlay2/8c5a99edd8e7c735f1ba5c0bbf003969046333538ec07d6f4f372effd2f42e16/merged]\\nnvidia-container-cli: mount error: stat failed: /proc/driver/nvidia/gpus/c060:00:00.0: no such file or directory\\n\"".
ERRO[0128] error getting events from daemon: context canceled
```

**Comment by RenaudWasTaken** [comment]
Hello,

This was fixed starting from 396.57

Closing this bug!
